### DISCUSSION TITLE
Livetiming output format (?)

### DISCUSSION THREAD

**Discussion by marinik98** [theOehrly/Fast-F1#95] [Q&A]
Hi, I wanted to integrate your livetiming module into something automated and I was wondering if we can have samples of those outputs, so we can maybe work with them. Thanks!

**Comment by theOehrly** [comment]
Here's a full recording of Practice 1 of last year's Portuguese Grand Prix.


[livetiming_p1_portugese_gp_2021.txt](https://github.com/theOehrly/Fast-F1/files/7883794/livetiming_p1_portugese_gp_2021.txt)


The data format is mostly the same as received when downloading from the API after a session. But there are some minor deviations. I'm really only "fixing" these slight differences (this is done when loading data in the LiveTimingData object) before passing it through the api parser. Note though, that the api parser in fastf1 cannot properly process partial data. Some look ahead and look back logic ensures that all the lap timing related data is mapped to the correct laps. 

To load this data in FastF1 you'd do the following:

```
import fastf1
from fastf1.livetiming.data import LiveTimingData

data = LiveTimingData('./livetiming_p1_portugese_gp_2021.txt')

session = fastf1.get_session(2021, 'Portugal', 'FP1')
laps = session.load_laps(livedata=data, with_telemetry=True)
```
(Also use caching though)
